### PR TITLE
Support for multiple discriminators on single type

### PR DESCRIPTION
**Pull request by zawodskoj**
This PR allows to add multiple discriminator values for single subtype which can cover all of discriminators. This is required for compound discriminators (tuples for example) and allows to write converters like this
```
JsonSubtypesConverterBuilder
    .Of(typeof(DeviceEvent), nameof(DeviceEvent.Discriminator))
    .RegisterSubtype(typeof(MetalDetectingAlert), (EventType.ALERT, DeviceType.METAL_DETECTOR))
    .RegisterSubtype(typeof(DeviceFailureWarning), (EventType.WARNING, DeviceType.METAL_DETECTOR))
    .RegisterSubtype(typeof(DeviceFailureWarning), (EventType.WARNING, DeviceType.CAMERA))
    .Build()
```

### CODE REVIEW

**Comment by manuc66** [comment]
Hi,

Thanks for submitting a PR to this project

Could you please add some unit tests with serialization and deserialization ?

And don't worry about the build failure related to SonarQube (it's a configuration problem)

File: JsonSubTypes/JsonSubtypesConverter.cs [hunk 1]
```diff
@@ -47,7 +47,19 @@ internal class JsonSubtypesConverter : JsonSubtypes
             _addDiscriminatorFirst = addDiscriminatorFirst;
             foreach (var type in _subTypeMapping)
             {
-                _supportedTypes.Add(type.Value, type.Key);
+                if (_supportedTypes.ContainsKey(type.Value))
+                {
+                    if (_addDiscriminatorFirst)
```
Comment (manuc66) [inline JsonSubTypes/JsonSubtypesConverter.cs]: `_addDiscriminatorFirst` or `_serializeDiscriminatorProperty` ?

File: JsonSubTypes/JsonSubtypesConverter.cs [hunk 2]
```diff
@@ -47,7 +47,19 @@ internal class JsonSubtypesConverter : JsonSubtypes
             _addDiscriminatorFirst = addDiscriminatorFirst;
             foreach (var type in _subTypeMapping)
             {
-                _supportedTypes.Add(type.Value, type.Key);
+                if (_supportedTypes.ContainsKey(type.Value))
```
Comment (manuc66) [inline JsonSubTypes/JsonSubtypesConverter.cs]: This kind of check is better suited in `JsonSubtypesConverterBuilder` (this class is internal)

File: JsonSubTypes/JsonSubtypesConverter.cs (hunk 1 above)
Comment (zawodskoj) [reply JsonSubTypes/JsonSubtypesConverter.cs]: Yes, it should be _serializeDiscriminatorProperty

File: JsonSubTypes/JsonSubtypesConverter.cs (hunk 2 above)
Comment (zawodskoj) [reply JsonSubTypes/JsonSubtypesConverter.cs]: ConverterBuilder contains inverted dictionary, but we can add check for it anyway

**Comment by zawodskoj** [comment]
It should work with KnownSubTypeAttribute. Attributed converter does not serialize discriminator and does not require type-to-discriminator mapping

I can add tests later

**Pull request merged**